packages/graphback-runtime-knex/tests/data/KnexDbDataProviderTest.ts

```diff
@@ -736,3 +732,379 @@ type Todo {
+      or: [
+        {
+          a: {
+            eq: 1
+          },
+          or: [
+            {
+              c: {
+                eq: 6
+              }
+            }, {
+              b: {
+                eq: 5
+              }
+            }
+          ]
+        }
+      ]
+    }
+
+    const items = await Todo.findBy({ filter });
+
+    expect(items).toHaveLength(2);
+  });
+
+  it('a || a || a', async () => {
+    const { providers: { Todo } } = await setup(`
+    
+
+    """
+    @model
+    """
+    type Todo {
+      id: ID
+      a: Int
+      b: Int
+      c: Int
+    }
+    `, {
+      seedData: {
+        todo: [
+          {
+            a: 1,
+            b: 5,
+            c: 8
+          },
+          {
+            a: 2,
+            b: 2,
+            c: 10
+          },
+          {
+            a: 3,
+            b: 5,
+            c: 3
+          },
+          {
+            a: 6,
+            b: 6,
+            c: 3
+          }
+        ]
+      }
+    });
+
+    const filter: QueryFilter = {
+      or: [
+        {
+          a: {
+            eq: 1
+          }
+        },
+        {
+          a: {
+            eq: 2
+          }
+        },
+        {
+          a: {
+            eq: 3
+          }
+        }
+      ]
+    }
+
+    const items = await Todo.findBy({ filter });
+
+    expect(items).toHaveLength(3)
+  })
+
+  it('a || (a && b)', async () => {
+    const { providers: { Todo } } = await setup(`
+    """
+    @model
+    """
+    type Todo {
+      id: ID
+      a: Int
+      b: Int
+      c: Int
+    }
+    `, {
+      seedData: {
+        todo: [
+          {
+            a: 1,
+            b: 5,
+            c: 8
+          },
+          {
+            a: 2,
+            b: 3,
+            c: 10
+          },
+          {
+            a: 2,
+            b: 3,
+            c: 3
+          },
+          {
+            a: 6,
+            b: 6,
+            c: 3
+          }
+        ]
+      }
+    });
+
+    const filter: QueryFilter = {
+      or: [
+        {
+          a: {
+            eq: 1
+          },
+        },
+        {
+          a: {
+            eq: 2
+          },
+          b: {
+            eq: 3
+          }
+        }
+      ]
+    }
+
+    const items = await Todo.findBy({ filter });
+
+    expect(items).toHaveLength(3)
+  })
+})
+
+test('or > not', async () => {
+
+  const { providers: { Todo } } = await setup(`
+    """
+    @model
+    """
+    type Todo {
+      id: ID
+      a: Int
+    }
+    `, {
+    seedData: {
+      todo: [
+        {
+          a: 1,
+        },
+        {
+          a: 2,
+        },
+        {
+          a: 2,
+        },
+        {
+          a: 3,
+        },
+        {
+          a: 11
+        }
+      ]
+    }
+  });
+
+  const filter: QueryFilter = {
+    or: [
+      {
+        a: {
+          in: [1, 11]
+        }
+      },
+      {
+        not: {
+          a: {
+            eq: 2
+          }
+        }
+      }
+    ]
+  }
```

Comment:
If I read this correctly, this equals: `either "a" in [1, 11] or "a" not equal 2`, right? If so, should the expectation be 
`expect(items).toHaveLength(3)` since `{a: 3}` satisfies the condition too?

Reply:
Yes - I read it wrong and it confused me, looks like this one is wrong!

Reply:
I suppose it's the same for Mongo? (I've not gotten to that part yet)

Reply:
The tests are different so it may not be the case, Mongo might work fine. I'm hitting difficulty with this one, and it is really to do with the arrangement of the SQL operators with Knex - so I'd say the issue is not present in Mongo.

Reply:
Tried this out for Mongo and indeed it's not the case.  So looks like a Knex only issue.

Reply:
Fixed now in https://github.com/aerogear/graphback/pull/2034/commits/9cbb4f0cdb61aac4f27a59c04ea89b7c9b63cb9b